pkg/istio-agent/health/health_probers.go

```diff
@@ -76,6 +76,10 @@ func NewHTTPProber(cfg *v1alpha3.HTTPHealthCheckConfig) *HTTPProber {
 			DisableKeepAlives: true,
 		}
 	}
+	d := &net.Dialer{
+		LocalAddr: &net.TCPAddr{IP: net.ParseIP("127.0.0.6")},
```

Comment:
I think we need the ipv6 conditional logic like for the prober?

Reply:
and possibly conditional on the legacy flag? Whatever we do with the rewrite probe should be consistent

Reply:
In https://github.com/istio/istio/pull/31121, it looks like only the destination of the probe depends on the flag.